File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/JMBA_InComm_iOS_SDK/InCommSDK/Models/Request/InCommOrderPurchaser.swift
//
//  InCommOrderPurchaser.swift
//  InCommSDK
//
//  Created by Taha Samad on 8/18/15.
//  Copyright (c) 2015 Fishbowl. All rights reserved.
//

import Foundation
import SwiftyJSON

public struct InCommOrderPurchaser {

    public var country: String
    public var emailAddress: String
    public var firstName: String
    public var lastName: String
    //
    public var city: String?
    public var companyName: String?
    public var mobilePhoneNumber: String?
    public var phoneNumber: String?
    public var sendDeliveryEmailAlert: Bool?
    public var sendDeliveryTextAlert: Bool?
    public var sendViewEmailAlert: Bool?
    public var sendViewTextAlert: Bool?
    public var stateProvince: String?
    public var streetAddress1: String?
    public var streetAddress2: String?
    public var suppressReceiptEmail: Bool?
    public var userId: Int32?
    public var zipPostalCode: String?
    
    public init(firstName: String, lastName: String, emailAddress: String, country: String) {
        self.firstName    = firstName
        self.lastName     = lastName
        self.emailAddress = emailAddress
        self.country      = country
    }
    
    public func serializeAsJSONDictionary() -> InCommJSONDictionary {
        var jsonDict                         = InCommJSONDictionary()
        jsonDict["Country"]                  = country
        jsonDict["EmailAddress"]             = emailAddress
        jsonDict["FirstName"]                = firstName
        jsonDict["LastName"]                 = lastName
        //
        jsonDict["City"]                     = city
        jsonDict["CompanyName"]              = companyName
        jsonDict["MobilePhoneNumber"]        = mobilePhoneNumber
        jsonDict["PhoneNumber"]              = phoneNumber
        jsonDict["SendDeliveryEmailAlert"]   = sendDeliveryEmailAlert
        jsonDict["SendDeliveryTextAlert"]    = sendDeliveryTextAlert
        jsonDict["SendViewEmailAlert"]       = sendViewEmailAlert
        jsonDict["SendViewTextAlert"]        = sendViewTextAlert
        jsonDict["StateProvince"]            = stateProvince
        jsonDict["StreetAddress1"]           = streetAddress1
        jsonDict["StreetAddress2"]           = streetAddress2
        jsonDict["SuppressReceiptEmail"]     = suppressReceiptEmail
        jsonDict["UserId"]                   = userId != nil ? NSNumber(int: userId!) : nil
        jsonDict["ZipPostalCode"]            = zipPostalCode
        return jsonDict
    }
    
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.2/JMBA_InComm_Android_SDK/app/src/main/java/com/wearehathway/apps/incomm/Models/InCommOrderPurchaser.java
package com.wearehathway.apps.incomm.Models;

/**
 * Created by Nauman Afzaal on 13/08/15.
 */
public class InCommOrderPurchaser
{
    private String City;
    private String CompanyName;
    private String Country;
    private String EmailAddress;
    private String FirstName;
    private String LastName;
    private String MobilePhoneNumber;
    private String PhoneNumber;
    private boolean SendDeliveryEmailAlert; //Valid values are "true" and "false." If true, the purchaser will receive an email once the gift card is delivered to the recipient.
    private boolean SendDeliveryTextAlert; //Valid values are "true" and "false." If true, the purchaser will receive a text once the gift card is delivered to the recipient.
    private boolean SendViewEmailAlert; //Valid values are "true" and "false." If true, the purchaser will receive an email once the gift card is viewed by the recipient.
    private boolean SendViewTextAlert; //Valid values are "true" and "false." If true, the purchaser will receive a text message once the gift card is viewed by the recipient.
    private String StateProvince;
    private String StreetAddress1;
    private String StreetAddress2;
    private boolean SuppressReceiptEmail;//Valid values are "true" and "false." If true, the purchaser will receive an email receipt.
    private String ZipPostalCode;

    public String getCity()
    {
        return City;
    }

    public void setCity(String city)
    {
        City = city;
    }

    public String getCompanyName()
    {
        return CompanyName;
    }

    public void setCompanyName(String companyName)
    {
        CompanyName = companyName;
    }

    public String getCountry()
    {
        return Country;
    }

    public void setCountry(String country)
    {
        Country = country;
    }

    public String getEmailAddress()
    {
        return EmailAddress;
    }

    public void setEmailAddress(String emailAddress)
    {
        EmailAddress = emailAddress;
    }

    public String getFirstName()
    {
        return FirstName;
    }

    public void setFirstName(String firstName)
    {
        FirstName = firstName;
    }

    public String getLastName()
    {
        return LastName;
    }

    public void setLastName(String lastName)
    {
        LastName = lastName;
    }

    public String getMobilePhoneNumber()
    {
        return MobilePhoneNumber;
    }

    public void setMobilePhoneNumber(String mobilePhoneNumber)
    {
        MobilePhoneNumber = mobilePhoneNumber;
    }

    public String getPhoneNumber()
    {
        return PhoneNumber;
    }

    public void setPhoneNumber(String phoneNumber)
    {
        PhoneNumber = phoneNumber;
    }

    public boolean isSendDeliveryEmailAlert()
    {
        return SendDeliveryEmailAlert;
    }

    public void setSendDeliveryEmailAlert(boolean sendDeliveryEmailAlert)
    {
        SendDeliveryEmailAlert = sendDeliveryEmailAlert;
    }

    public boolean isSendDeliveryTextAlert()
    {
        return SendDeliveryTextAlert;
    }

    public void setSendDeliveryTextAlert(boolean sendDeliveryTextAlert)
    {
        SendDeliveryTextAlert = sendDeliveryTextAlert;
    }

    public boolean isSendViewEmailAlert()
    {
        return SendViewEmailAlert;
    }

    public void setSendViewEmailAlert(boolean sendViewEmailAlert)
    {
        SendViewEmailAlert = sendViewEmailAlert;
    }

    public boolean isSendViewTextAlert()
    {
        return SendViewTextAlert;
    }

    public void setSendViewTextAlert(boolean sendViewTextAlert)
    {
        SendViewTextAlert = sendViewTextAlert;
    }

    public String getStateProvince()
    {
        return StateProvince;
    }

    public void setStateProvince(String stateProvince)
    {
        StateProvince = stateProvince;
    }

    public String getStreetAddress1()
    {
        return StreetAddress1;
    }

    public void setStreetAddress1(String streetAddress1)
    {
        StreetAddress1 = streetAddress1;
    }

    public String getStreetAddress2()
    {
        return StreetAddress2;
    }

    public void setStreetAddress2(String streetAddress2)
    {
        StreetAddress2 = streetAddress2;
    }

    public boolean isSuppressReceiptEmail()
    {
        return SuppressReceiptEmail;
    }

    public void setSuppressReceiptEmail(boolean suppressReceiptEmail)
    {
        SuppressReceiptEmail = suppressReceiptEmail;
    }

    public String getZipPostalCode()
    {
        return ZipPostalCode;
    }

    public void setZipPostalCode(String zipPostalCode)
    {
        ZipPostalCode = zipPostalCode;
    }
}
